opentaxii/taxii2/utils.py

```diff
@@ -15,3 +15,31 @@ def taxii2_datetimeformat(input_value: datetime.datetime) -> str:
     :rtype: string
     """
     return input_value.astimezone(datetime.timezone.utc).strftime(DATETIMEFORMAT)
+
+
+def get_object_version(obj: dict) -> datetime.datetime:
+    """
+    The TAXII version is inferred with this order:
+
+    1. modified field if present
+    2. created field if present
+    3. 1970-01-01T00:00:00+00:00
+
+        If a STIX object is not versioned (and therefore does not have a modified
+        timestamp) then this version parameter MUST use the created timestamp. If
+        an object does not have a created or modified timestamp or any other
+        version information that can be used, then the server should use a value for
+        the version that is consistent to the server.
+
+    -- TAXII 2.1 specification --
```

Comment:
We can use normal reST footnote:

```suggestion
    The TAXII version is inferred in this order:

    1. ``modified`` field, if present
    2. ``created`` field, if present
    3. 1970-01-01T00:00:00+00:00 otherwise
    
    This is guided by 3.4.1 Supported Fields for Match [#]_: 

        If a STIX object is not versioned (and therefore does not have a modified
        timestamp) then this version parameter MUST use the created timestamp. If
        an object does not have a created or modified timestamp or any other
        version information that can be used, then the server should use a value for
        the version that is consistent to the server.

    .. [#] https://docs.oasis-open.org/cti/taxii/v2.1/os/taxii-v2.1-os.html#_Toc31107518
```